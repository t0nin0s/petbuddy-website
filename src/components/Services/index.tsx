import React from 'react';
import styled from 'styled-components';
import Button from '../Button'
import { Services } from '../../types'

const Container = styled.div`
display: flex;
flex-direction: column;
letter-spacing: 2px;
line-height: 1.8em;
align-items: center;
padding:2rem;`

const Img = styled.img`
width:85%;
border-radius:10px
`
const Title = styled.h4`
font-size:24px
`
const Description = styled.div`
margin-bottom:2rem
`

export default (props: Services) => {
  const { img, header, description, ctaText } = props
  return (
    <Container>
      <Img src={img} alt="Service image" />
      <Title> {header}</Title>
      <Description>{description}</Description>
      <Button>{ctaText}</Button>
    </Container>
  )
}