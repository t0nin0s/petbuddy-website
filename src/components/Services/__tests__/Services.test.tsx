import React from 'react';
import { render } from '@testing-library/react';
import Services from '../index'

describe('Services component', () => {
  it('should render', () => {
    const wrapper = render(<Services />)
    expect(wrapper).toMatchSnapshot();
  })
  it('should output services title', () => {
    const wrapper = 'Dog Services'
    expect(wrapper).toEqual('Dog Services');
  })
})
