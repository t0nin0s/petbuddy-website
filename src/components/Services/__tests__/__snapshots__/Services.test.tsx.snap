// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsTCUz JUeOo"
      >
        <img
          alt="Service image"
          class="sc-dlfnbm cDLTwB"
        />
        <h3
          class="sc-hKgILt"
        >
           
        </h3>
        <p
          class="sc-eCssSg imylDa"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdfBwQ jdtooK"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsTCUz JUeOo"
    >
      <img
        alt="Service image"
        class="sc-dlfnbm cDLTwB"
      />
      <h3
        class="sc-hKgILt"
      >
         
      </h3>
      <p
        class="sc-eCssSg imylDa"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdfBwQ jdtooK"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
